internal/helmchart/settings.go

```diff
@@ -0,0 +1,55 @@
+// Copyright © 2021 - 2023 SUSE LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helmchart
+
+import (
+	"errors"
+
+	"github.com/epinio/epinio/pkg/api/core/v1/models"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func SettingsToChart(chart *unstructured.Unstructured) (map[string]models.ChartSetting, error) {
```

Comment:
I like the conversion from the unstructured object instead of using them in the code (I have seen that), but when we know the type we can use directly the kubernetes provided converter.

See

https://github.com/epinio/epinio/blob/bb5622333bd49c5ffa0d6edd32aeed08f8a12e22/internal/services/catalog.go#L84-L88

Reply:
Will have a look. Side note: We use a lot of explicit unstructured conversion for both app(charts) and service catalog, I believe. Should look into general simplification ...

Reply:
Done with commit fed941e7a2e4b1ac1d3e40af08ddfa6a6f9a9e47
Note: Left the appchart code untouched, i.e. this is still based on unstructured